src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -568,24 +549,15 @@ private BigInteger(uint[] value)
             if (!isNegative)
             {
                 // Handle the simple positive value cases where the input is already in sign magnitude
-                if (dwordCount != value.Length)
-                {
-                    _sign = +1;
-                    _bits = new uint[dwordCount];
-                    Array.Copy(value, _bits, dwordCount);
-                }
-                // No trimming is possible.  Assign value directly to _bits.
-                else
-                {
-                    _sign = +1;
-                    _bits = value;
-                }
+                _sign = +1;
+                value = value.Slice(0, dwordCount);
```

Comment:
same here, this can just be `value.Slice().ToArray()`

Reply:
and any other cases where the same pattern is used (I see at least one other place below)

Reply:
Fixed